aws/resource_aws_sns_topic_subscription_test.go

```diff
@@ -245,3 +279,204 @@ resource "aws_sns_topic_subscription" "test_subscription" {
+  resource_id = "${aws_api_gateway_rest_api.test.root_resource_id}"
+  http_method = "${aws_api_gateway_method.test.http_method}"
+  status_code = "${aws_api_gateway_method_response.test.status_code}"
+
+  response_parameters {
+    "method.response.header.Access-Control-Allow-Origin" = "'*'"
+  }
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name = "tf-acc-test-sns-%d"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "policy" {
+  name = "tf-acc-test-sns-%d"
+  role = "${aws_iam_role.iam_for_lambda.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "logs:*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_permission" "apigw_lambda" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.lambda.arn}"
+  principal     = "apigateway.amazonaws.com"
+  source_arn    = "${aws_api_gateway_deployment.test.execution_arn}/*"
+}
+
+resource "aws_lambda_function" "lambda" {
+  filename         = "test-fixtures/lambda_confirm_sns.zip"
+  function_name    = "tf-acc-test-sns-%d"
+  role             = "${aws_iam_role.iam_for_lambda.arn}"
+  handler          = "main.confirm_subscription"
+  source_code_hash = "${base64sha256(file("test-fixtures/lambda_confirm_sns.zip"))}"
+  runtime          = "python3.6"
+}
+
+resource "aws_api_gateway_deployment" "test" {
+  depends_on  = ["aws_api_gateway_integration_response.test"]
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  stage_name  = "acctest"
+}
+
+resource "aws_iam_role" "invocation_role" {
+  name = "tf-acc-test-authorizer-%d"
+  path = "/"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "apigateway.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "invocation_policy" {
+  name = "tf-acc-test-authorizer-%d"
+  role = "${aws_iam_role.invocation_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "lambda:InvokeFunction",
+      "Effect": "Allow",
+      "Resource": "${aws_lambda_function.authorizer.arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_api_gateway_authorizer" "test" {
+  name                             = "tf-acc-test-api-gw-authorizer-%d"
+  rest_api_id                      = "${aws_api_gateway_rest_api.test.id}"
+  authorizer_uri                   = "arn:aws:apigateway:eu-west-1:lambda:path/2015-03-31/functions/${aws_lambda_function.authorizer.arn}/invocations"
```

Comment:
Do you mind changing this to

```
${aws_lambda_function.authorizer.invoke_arn}
```
? That will make it a bit easier to read and (more importantly) make it work in `us-west-2` region which we use for acceptance tests 😉